python/MooseDocs/extensions/listing.py

```diff
@@ -86,6 +120,130 @@ def createToken(self, parent, info, page, settings):
 
         return parent
 
+    def appendMooseSyntax(self, content: str, page: pages.Page) -> str:
+        """
+        Append content with syntax/object/parameter information.
+
+        This method will utilize pyhit to get syntaxes contained in the input
+        then use the appsyntax extension to associate those syntaxes with nodes
+        containing the object descriptions and documentation location. Once that
+        node is located, it appends via json format that data.
+
+        An example would be::
+
+            [Mesh] -> [Mesh<<<{'href': '...'}>>>]
+               [gmg]
+                 type = GeneratedMeshGenerator -> type = GeneratedMeshGenerator<<<{'description': '...', 'href': '...'}>>>
+                 dim = 2 -> dim<<<{'description': '...'}>>> = 2
+               []
+            []
+
+        The `framework/doc/content/contrib/prism/prism.min.js` JavaScript file
+        knows to recognize the `<<<>>>` syntax and
+        `framework/doc/content/js/moose_input_parser.js` JavaScript file knows
+        how to add the approriate HTML links and tooltips.
+        """
+        if self.extension.syntax is None:
+            return content
+
+        syntax_ext: appsyntax.AppSyntaxExtension = self.extension.syntax # Convenience
+
+        content_lines = content.splitlines()
+        try:
+            root = pyhit.parse(content)
+        except:
```

Comment:
Is it worth adding a warning here to see if this is a common issue?

Reply:
I don't think so. I think it is too common that people put in snippets that are invalid according to pyhit. And example would be something like this:
```
[Mesh]
   [gmg]
       type = GeneratedMeshGenerator
       ...
   []
[]
```

This should probably be allowed without warnings.